src/process/posix/SDL_posixprocess.c

```diff
@@ -0,0 +1,370 @@
+{
+    SDL_Process *process = (SDL_Process *) userdata;
+
+    ssize_t ret = read(process->internal->stdout_pipe, ptr, size);
+
+    if (ret < 0) {
+        if (status) {
+            *status = SDL_IO_STATUS_ERROR;
+        }
+        SDL_SetError("Could not read(): %s", strerror(errno));
+        return 0;
+    }
+
+    return ret;
+}
+
+static size_t SDLCALL process_read_from_stderr(void *userdata, void *ptr, size_t size, SDL_IOStatus *status)
+{
+    SDL_Process *process = (SDL_Process *) userdata;
+
+    ssize_t ret = read(process->internal->stderr_pipe, ptr, size);
+
+    if (ret < 0) {
+        if (status) {
+            *status = SDL_IO_STATUS_ERROR;
+        }
+        SDL_SetError("Could not read(): %s", strerror(errno));
+        return 0;
+    }
+
+    return ret;
+}
+
+static size_t SDLCALL process_write_to_stdin(void *userdata, const void *ptr, size_t size, SDL_IOStatus *status)
+{
+    SDL_Process *process = (SDL_Process *) userdata;
+
+    ssize_t ret = write(process->internal->stdin_pipe, ptr, size);
+
+    if (ret < 0) {
+        SDL_SetError("Could not write(): %s", strerror(errno));
+        if (status) {
+            *status = SDL_IO_STATUS_ERROR;
+        }
+        return 0;
+    }
+
+    return ret;
+}
+
+static SDL_bool SDLCALL process_stdin_close(void *userdata)
+{
+    SDL_Process *process = (SDL_Process *) userdata;
+
+    close(process->internal->stdin_pipe);
+    SDL_ClearProperty(process->props, SDL_PROP_PROCESS_STDIN_POINTER);
+    return true;
+}
+
+static SDL_bool SDLCALL process_stdout_close(void *userdata)
+{
+    SDL_Process *process = (SDL_Process *) userdata;
+
+    close(process->internal->stdout_pipe);
+    SDL_ClearProperty(process->props, SDL_PROP_PROCESS_STDOUT_POINTER);
+    return true;
+}
+
+static SDL_bool SDLCALL process_stderr_close(void *userdata)
+{
+    SDL_Process *process = (SDL_Process *) userdata;
+
+    close(process->internal->stderr_pipe);
+    SDL_ClearProperty(process->props, SDL_PROP_PROCESS_STDERR_POINTER);
+    return true;
+}
+
+bool SDL_SYS_CreateProcessWithProperties(SDL_Process *process, SDL_PropertiesID props)
+{
+    char * const *args = SDL_GetPointerProperty(props, SDL_PROP_PROCESS_CREATE_ARGS_POINTER, NULL);
+    char * const *env = SDL_GetPointerProperty(props, SDL_PROP_PROCESS_CREATE_ENVIRONMENT_POINTER, NULL);
+    bool enable_stdin = SDL_GetBooleanProperty(props, SDL_PROP_PROCESS_CREATE_STDIN_BOOLEAN, false);
+    bool enable_stdout = SDL_GetBooleanProperty(props, SDL_PROP_PROCESS_CREATE_STDOUT_BOOLEAN, false);
+    bool enable_stderr = SDL_GetBooleanProperty(props, SDL_PROP_PROCESS_CREATE_STDERR_BOOLEAN, false);
+    bool redirect_stderr = SDL_GetBooleanProperty(props, SDL_PROP_PROCESS_CREATE_REDIRECT_STDERR_BOOLEAN, false);
+    int stdin_pipe[2] = { -1, -1 };
+    int stdout_pipe[2] = { -1, -1 };
+    int stderr_pipe[2] = { -1, -1 };
+
+    // Keep the malloc() before exec() so that an OOM won't run a process at all
```

Comment:
Avoid dereferencing a null pointer when calling `posix_spawnp`.is `NULL`.
I'm unsure whether this should also test `!args[0]`
```suggestion
    int stderr_pipe[2] = { -1, -1 };

   if (!args) {
       return false;
   }

    // Keep the malloc() before exec() so that an OOM won't run a process at all
```

Reply:
args is guaranteed to be valid at this point.